tests/e2e/test_rosacli_iam_roles.go

```diff
@@ -874,4 +874,130 @@ var _ = Describe("Edit IAM",
 				foundUserRole = userRoleList.UserRole(userrolePrefix, ocmAccountUsername)
 				Expect(foundUserRole.Linded).To(Equal("No"))
 			})
+
+		It("create/delete hypershift account roles with managed policies - [id:61322]",
+			labels.Critical,
+			func() {
+				defer func() {
```

Comment:
Just curious as to why we use defer over using `AfterEach` or `AfterSuite`?

Reply:
Actually we use cluster.CleanResources() func in after suite, but in case of roles different test cases needs to delete different roles like account-roles, operator-roles, etc, so in this ile we use actually defer func invidually for specific cases